### PR TITLE
Finding files to parse using wildcards

### PR DESCRIPTION
**Pull request by kgromadzki**

### CODE REVIEW

File: whylog/config/log_type.py [hunk 1]
```diff
@@ -1,21 +1,12 @@
...
-            (
-                'localhost', os.path.join(
-                    *[
-                        'whylog', 'tests', 'tests_log_reader', 'test_files', '003_match_time_range',
-                        'node_1.log'
-                    ]
-                )
-            )
-        ]  # yapf: disable
+        parsed_files = set()
```
Comment (konefalg) [inline whylog/config/log_type.py:11]: Please add docstring describing what does this function do. It's not obvius.

File: whylog/config/filename_matchers.py [hunk 2]
```diff
@@ -7,25 +7,32 @@
...
-class RegexFilenameMatcher(AbstractFilenameMatcher):
+class WildCardFilenameMatcher(AbstractFilenameMatcher):
     def __init__(self, host_pattern, path_pattern, log_type_name):
         self.log_type_name = log_type_name
-        self._path_pattern = path_pattern
-        self._host_pattern = host_pattern
-        self._host_regex = re.compile(host_pattern)
-        self._path_regex = re.compile(path_pattern)
+        self.path_pattern = path_pattern
+        self.host_pattern = host_pattern
```
Comment (konefalg) [inline whylog/config/filename_matchers.py:16]: Please keep class fields order the same as in constructor params.

File: whylog/tests/utils.py [hunk 3]
```diff
@@ -1,10 +1,17 @@
 import os.path
+import platform
 
 
 class ConfigPathFactory(object):
     @classmethod
-    def get_path_to_config_files(cls, prefix_path):
+    def get_path_to_config_files(cls, prefix_path, multi_platform=False):
```
Comment (konefalg) [inline whylog/tests/utils.py]: Why it's False by default?

File: whylog/config/log_type.py [hunk 4]
```diff
@@ -1,21 +1,16 @@
...
-                    *[
-                        'whylog', 'tests', 'tests_log_reader', 'test_files', '003_match_time_range',
-                        'node_1.log'
-                    ]
-                )
-            )
-        ]  # yapf: disable
+        """
+        Gets all possible distinct pairs (host, file_name) belongs to single log type
+        It's information for LogReader which files should be parsed by its.
```
Comment (ppolewicz) [inline whylog/config/log_type.py]: By its what?

File: whylog/config/log_type.py [hunk 5]
```diff
@@ -1,21 +1,16 @@
...
-                'localhost', os.path.join(
-                    *[
-                        'whylog', 'tests', 'tests_log_reader', 'test_files', '003_match_time_range',
-                        'node_1.log'
-                    ]
-                )
-            )
-        ]  # yapf: disable
+        """
+        Gets all possible distinct pairs (host, file_name) belongs to single log type
```
Comment (ppolewicz) [inline whylog/config/log_type.py]: s/belongs/belonging/

File: whylog/tests/utils.py [hunk 6]
```diff
@@ -1,10 +1,17 @@
...
 class ConfigPathFactory(object):
     @classmethod
-    def get_path_to_config_files(cls, prefix_path):
+    def get_path_to_config_files(cls, prefix_path, multi_platform=True):
         parsers_path = os.path.join(prefix_path, 'parsers.yaml')
         rules_path = os.path.join(prefix_path, 'rules.yaml')
-        log_type_path = os.path.join(prefix_path, 'log_types.yaml')
+        if multi_platform:
+            if platform.system() == 'Windows':
+                log_type_path = os.path.join(prefix_path, 'windows_log_types.yaml')
```
Comment (ppolewicz) [inline whylog/tests/utils.py]: Extract the string to a variable and then DRY on the rest of the line. Also please add a comment which will say what is going on here and why.

**Pull request merged**